src/main/java/core/basesyntax/service/impl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ReportGenerator;
+import java.util.Map;
+import java.util.Set;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    @Override
+    public String getReport(Storage storage) {
+        Map<String, Integer> fruitMap = storage.getAll();
+        Set<String> fruits = fruitMap.keySet();
+        StringBuilder sb = new StringBuilder();
+        for (String fruit : fruits) {
+            sb.append("\n").append(fruit).append(fruitMap.get(fruit));
```

Comment:
The format of the report is not specified in the checklist, but typically, a report would have some kind of delimiter between the fruit and its quantity, such as a comma or a colon. Consider adding one for clarity.